cudapoa/include/claragenomics/cudapoa/batch.hpp

```diff
@@ -44,6 +45,60 @@ struct Entry
 /// A type defining the set and order of Entry's in which a POA is processed.
 typedef std::vector<Entry> Group;
 
+/// A structure to hold  upper limits for data size processed in POA batches
+struct BatchSize
+{
+    /// Maximum number of elements in a sequence
+    uint32_t max_sequence_size = 1024;
+    /// Maximum size of final consensus
+    uint32_t max_concensus_size = 1024;
+    /// Maximum number of nodes in a graph, 1 graph per window
+    uint32_t max_nodes_per_window = 3072;
+    /// Maximum number of nodes in a graph, 1 graph per window in banded mode
+    uint32_t max_nodes_per_window_banded = 4096;
+    /// Maximum vertical dimension of scoring matrix, which stores graph
+    /// Adding 4 elements more to ensure a 4byte boundary alignment for any allocated buffer
+    uint32_t max_matrix_graph_dimension = max_nodes_per_window + 4;
+    /// Maximum vertical dimension of scoring matrix, which stores graph
+    /// Adding 4 elements more to ensure a 4byte boundary alignment for any allocated buffer
+    uint32_t max_matrix_graph_dimension_banded = max_nodes_per_window_banded + 4;
+    /// Maximum horizontal dimension of scoring matrix, which stores sequences
+    /// Adding 4 elements more to ensure a 4byte boundary alignment for any allocated buffer
+    uint32_t max_matrix_sequence_dimension = max_sequence_size + 4;
+    /// Maximum number of equences per POA group
+    uint32_t max_sequences_per_poa = 100;
+
+    /// set upper limit parameters based on max_sequence_size
+    void setSize(const uint32_t max_seq_sz, const uint32_t max_num_seqs)
```

Comment:
yes, before adding `setParams`, I was also thinking to use constructor instead of `setSize()`. will change it.

Reply:
done.

Reply:
I think it's better we add something like `throw std::invalid_argument` in case of using the constructors that takes all parameters is used and it violates some assumptions, like `max_sequence_size <= max_graph_size`, etc. what do you think @tijyojwad ?

Reply:
Yes that would be really good!

Reply:
I added few checks, but it's difficult to cover all invalid scenarios.